### PR TITLE
feat(quant): add quantized retinanet

### PR DESCRIPTION
**Pull request by FateScript**
add quantized retinanet under quant dir

please reveiew @wjfwzzc

### CODE REVIEW

File: official/quantization/retinanet/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import megengine.jit as jit
+
+model = ...
+
+Q.quantize_qat(model, qconfig=Q.ema_fakequant_qconfig)
+
+# real quant
+Q.quantize(model)
+
+@jit.trace(symbolic=True):
```
Comment (wjfwzzc) [inline official/quantization/retinanet/README.md:49]: typo

File: official/quantization/retinanet/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+模型量化 Model Quantization
+---
+
+本目录包含了采用MegEngine实现的检测模型量化训练和部署的代码，包括常用的RetinaNet，因为时间原因，仅仅量化了检测模型的Resnet-Backbone部分，后续会随着MegEngine的版本持续更新。
+
+| Model | mAP | test speed* (float32) | mAP (int8) | test speed* (int8) |
+| --- | --- | --- | --- | --- |
+| RetinaNet |  29.9  | 42it/s(8GPU)   | 29.7 | 45it/s(8GPU) |
```
Comment (wjfwzzc) [inline official/quantization/retinanet/README.md:8]: test speed报单卡比较好，另外应该报一下cpu的fp32和int8。

File: official/quantization/retinanet/loss.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
```
Comment (wjfwzzc) [inline official/quantization/retinanet/loss.py:1]: rebase一下master，这类冗余的文件可以删掉了

File: official/quantization/retinanet/fpn.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+        self.add_module.disable_quantize()
+        self.quant = M.QuantStub()
+        self.dequant = M.Sequential(*[M.DequantStub() for _ in range(5)])
+        # self.add = M.Elemwise("ADD")
+
+        # QAT
+        # for module in self.lateral_convs + self.output_convs:
+        #     module.disable_quantize()
+        # self.top_block.disable_quantize()
+        # self.add.disable_quantize()
```
Comment (wjfwzzc) [inline official/quantization/retinanet/fpn.py:138]: 这段注释代码如果一直都没用的话就删掉吧，如果要在某个环节用的话，最好是有个参数做if/else判断，否则看到这段话用户不清楚该不该用

File: official/quantization/retinanet/retinanet_res18.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+            labels_cat_list.append(F.add_axis(labels_cat, 0))
+            bbox_targets_list.append(F.add_axis(bbox_targets, 0))
+
+        return (
+            F.zero_grad(F.concat(labels_cat_list, axis=0)),
+            F.zero_grad(F.concat(bbox_targets_list, axis=0)),
+        )
+
+
+class RetinaNetConfig:
```
Comment (wjfwzzc) [inline official/quantization/retinanet/retinanet_res18.py:203]: 有带宽的话这个文件的net和config照现在master的状态refine一下，暂时不好弄的话就算了。

File: official/quantization/retinanet/train.py [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+    train_dataloader = DataLoader(
+        train_dataset,
+        sampler=train_sampler,
+        transform=T.Compose(
+            transforms=[
+                T.ShortestEdgeResize(
+                    cfg.train_image_short_size, cfg.train_image_max_size
+                ),
+                T.RandomHorizontalFlip(),
+                T.Normalize(mean=128),
```
Comment (wjfwzzc) [inline official/quantization/retinanet/train.py:291]: 这个吃进cfg的`img_mean`里

File: official/quantization/retinanet/train.py [hunk 7]
```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
```
Comment (wjfwzzc) [inline official/quantization/retinanet/train.py:1]: 这个文件也是，有带宽的话照现在master的train.py refine一下，尤其是一些工具函数已经可以直接用utils了

File: official/quantization/retinanet/test.py [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+        )
+        resized_img = cv2.resize(
+            img, (resized_width, resized_height), interpolation=cv2.INTER_LINEAR,
+        )
+        resized_img = cv2.flip(resized_img, 1) if flip else resized_img
+        trans_img = np.ascontiguousarray(
+            resized_img.transpose(2, 0, 1)[None, :, :, :], dtype=np.uint8
+        )
+        img_mean = np.array([128, 128, 128])
+        trans_img = trans_img - img_mean[None, :, None, None]
```
Comment (wjfwzzc) [inline official/quantization/retinanet/test.py:77]: 这个吃进cfg的`img_mean`里
